### PR TITLE
[React] Fix accepted types for some hooks

### PR DESCRIPTION
**Pull request by atharvadeosthale**
Hooks `useUnclaimedNFTs()`, `useClaimedNFTs()`, `useUnclaimedNFTSupply()`, and `useClaimedNFTSupply()` don't work with edition drop contracts, however, the comments on these hooks specified they could.

### CODE REVIEW

**Review by nachoiacovino — CHANGES_REQUESTED** [review]
we can't accept only NFTDrop which is the prebuilt, we need to accept DropContract (that has all drops) but exclude all erc1155 and erc20, so find how we do it elsewhere, but need to do a DropContract exclude EditionDrop, TokenDrop

**Comment by atharvadeosthale** [comment]
`useClaimedNFTs` and `useUnclaimedNFTs` took only `NFTDrop` even before so I changed their link to point to `NFTDrop`. For the ones where we exclude edition drop and token drop I've added normal text and removed links

File: packages/react-core/src/evm/hooks/async/drop.ts [hunk 1]
```diff
@@ -74,10 +76,10 @@ export function useUnclaimedNFTs(
  * const { data: claimedNFTs, isLoading, error } = useClaimedNFTs(contract, { start: 0, count: 100 });
  * ```
  *
- * @param contract - an instance of a {@link DropContract}
+ * @param contract - an instance of a {@link NFTDrop}
```
Comment (nachoiacovino) [inline packages/react-core/src/evm/hooks/async/drop.ts:79]: this is not correct now

**Review by nachoiacovino — CHANGES_REQUESTED** [review]
fix the params plz :D

**Review by nachoiacovino — APPROVED** [review]
lgtm!

**Pull request merged**